### ISSUE TITLE
Deep Deterministic Policy Gradient algorithm

### ISSUE THREAD

**Issue by sbenthall** [scikit-agent/scikit-agent#130]
There is a reinforcement learning (RL) algorithm used in robotic for problems with continuous action spaces, Deep Deterministic Policy Gradient (DDPG).

It is well documented as part of deep Multi-Agent Reinforcement Learning libraries and toolkits:
https://docs.pytorch.org/rl/main/tutorials/multiagent_competitive_ddpg.html
https://docs.pytorch.org/tutorials/advanced/coding_ddpg.html
https://github.com/ghliu/pytorch-ddpg
https://github.com/cookbenjamin/DDPG

This is an actor-critic method, meaning that it uses both a policy (actor) and value (critic) network, similar to the MMW method.
But it doesn't rely on either lifetime estimation, FOC conditions, or Envelope conditions.
Rather, it just experiences the game with a replay buffer and explores the action space with some noise.

This issue is for an implementation of this algorithm to solve the current suite of models.
We can use the BellmanPeriod object #121 #128 and the downstream neural network objects.

**Comment by sbenthall** [comment]
This is the paper that introduces the algorithm:
https://arxiv.org/abs/1509.02971

One thing this needs is a "Critic", which computes an estimated $Q(s,a)$ value. (As opposed to a value function $V(s)$.)

This can be done in `ann.py` as a `BlockQNet`. Similar architecture to the `BlockValueNet`.

DDPG explicit maintains, and updates separately, two different neural networks for actor and critic. I believe this is different from the Maliar algorithm, which updates both at once. But this difference might be more cosmetic than actual.

I bring this up because in discussions earlier this year, @alanlujan91 and I were going over whether the same training routine should be applied to Policy, Value, and PolicyValue networks. I now find it challenging to identify the precise differences and whether it matters. But it is probably best to stick to the papers for guidance.
